app/views/users/show.html.erb

```diff
@@ -24,15 +24,44 @@
     }
     <% end %>
   </style>
-  <div class="user-profile-header" style="<%= user_colors_style(@user) %>" itemscope itemtype="http://schema.org/Person" itemprop="mainEntity">
-    <meta itemprop="url" content="https://dev.to/<%= @user.username %>">
+
+<script type="application/ld+json">
+  {
+    "@context": "http://schema.org",
+    "@type": "Person",
+    "mainEntityOfPage": "True",
+    "url": "<%= user_url(@user) %>",
+    "image": <%= ProfileImage.new(@user).get(width: 320) %>,
+    "name": "<%= @user.name %>",
+    "description": "<%= @user.summary.presence || ["404 bio not found"].sample %>",
+    "email": "<%= @user.email %>",
+    "jobTitle": "<%= @user.employment_title %>",
+    "sameAs": [
+      "https://twitter.com/<%= @user.twitter_username %>",
+      "https://github.com/<%= @user.github_username %>",
+      "<%= @user.mastodon_url %>",
+      "<%= @user.facebook_url %>",
+      "<%= @user.linkedin_url %>",
+      "<%= @user.behance_url %>",
+      "<%= @user.stackoverflow_url %>",
+      "<%= @user.dribbble_url %>",
+      "<%= @user.medium_url %>",
+      "<%= @user.gitlab_url %>",
+      "<%= @user.instagram_url %>",
+      "<%= @user.twitch_url %>",
+      "<%= @user.website_url %>"
```

Comment:
these could all potentially be empty strings, probably checking ifs as Ben mentioned